src/vue/pages/SaleCreationRequests.vue

```diff
@@ -0,0 +1,359 @@
+                  {{ request.updatedAt | formatCalendar }}
+                </td>
+                <td>
+                  <a
+                    class="request-details-btn"
+                    @click="showRequestDetails(index)"
+                  >
+                    {{ 'sale-creation-requests.details-btn' | globalize }}
+                  </a>
+                </td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+      </template>
+
+      <template v-else>
+        <!-- eslint-disable max-len -->
+        <no-data-message
+          icon-name="trending-up"
+          :msg-title="'sale-creation-requests.no-request-history-title' | globalize"
+          :msg-message="'sale-creation-requests.no-request-history-desc' | globalize"
+        />
+        <!-- eslint-enable max-len -->
+      </template>
+    </template>
+
+    <template v-else-if="!isLoadingFailed">
+      <loader :message-id="'sale-creation-requests.loading-msg'" />
+    </template>
+
+    <template v-else>
+      <p>
+        {{ 'sale-creation-requests.loading-error-msg' | globalize }}
+      </p>
+    </template>
+
+    <collection-loader
+      class="sale-creation-requests__loader"
+      v-show="requestsHistory.length"
+      :first-page-loader="firstPageLoader"
+      @first-page-load="setHistory"
+      @next-page-load="extendHistory"
+      @error="isLoadingFailed = true"
+    />
+  </div>
+</template>
+
+<script>
+import Loader from '@/vue/common/Loader'
+import Drawer from '@/vue/common/Drawer'
+import NoDataMessage from '@/vue/common/NoDataMessage'
+import CollectionLoader from '@/vue/common/CollectionLoader'
+import SelectField from '@/vue/fields/SelectField'
+
+import CreateSaleForm from '@/vue/forms/CreateSaleForm'
+import SaleRequestDetails from '@/vue/pages/sales/SaleRequestDetails'
+
+import { Sdk } from '@/sdk'
+import { base } from '@tokend/js-sdk'
+
+import { RecordWrapper } from '@/js/records'
+
+import { mapGetters } from 'vuex'
+import { vuexTypes } from '@/vuex'
+
+import { Bus } from '@/js/helpers/event-bus'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+
+export default {
+  name: 'sale-creation-requests',
+  components: {
+    Loader,
+    Drawer,
+    NoDataMessage,
+    CollectionLoader,
+    SelectField,
+    CreateSaleForm,
+    SaleRequestDetails,
+  },
+
+  data: _ => ({
+    requestsHistory: [],
+    isLoaded: false,
+    isLoadingFailed: false,
+    isDetailsDrawerShown: false,
+    selectedIndex: -1,
+    isUpdateMode: false,
+    filters: {
+      baseAsset: '',
+    },
+    firstPageLoader: () => {},
+  }),
+  computed: {
+    ...mapGetters({
+      account: vuexTypes.account,
+    }),
+
+    selectedRequest () {
+      return this.filteredRequests[this.selectedIndex]
+    },
+
+    saleAssetCodes () {
+      return this.requestsHistory
+        .map(request => request.baseAsset)
+        .filter((asset, i, self) => self.indexOf(asset) === i)
+    },
+
+    filteredRequests () {
```

Comment:
Never filter pageable data

Reply:
Updated, used `OwnedAssetsLoaderMixin`, filtered in request